cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/saved_search.spec.js

```diff
@@ -0,0 +1,513 @@
+const allTestConfigurations = [
+  ...Object.values(indexPatternTestConfigurations),
+  ...Object.values(indexTestConfigurations),
+];
+
+// Maps a dataset that are used in this file to the exact string that dataset
+// corresponds to in the saved search API response
+const mapDatasetToType = (dataset) => {
+  switch (dataset) {
+    case INDEX_PATTERN_WITH_TIME:
+      return 'INDEX_PATTERN';
+    case INDEX_WITH_TIME_1:
+      return 'INDEXES';
+    default:
+      return 'unknown dataset';
+  }
+};
+
+// Maps a language that are used in this file to the exact string that the language
+// corresponds to in the saved search API response
+const mapLanguageToApiResponseString = (language) => {
+  switch (language) {
+    case 'DQL':
+      return 'kuery';
+    case 'Lucene':
+      return 'lucene';
+    case 'OpenSearch SQL':
+      return 'SQL';
+    case 'PPL':
+      return 'PPL';
+    default:
+      return 'unknown language';
+  }
+};
+
+// Escapes special characters for the editor
+const prepareQueryStringForEditor = (queryString) => {
+  return queryString.replaceAll(/([{}])/g, (char) => `{${char}}`);
+};
+
+// In theory this function is not needed as we should be able to use
+// the custom command navigateToWorkSpaceSpecificPage, but when using that
+// to go to the discover page, it sometimes leads to a blank page
+const navigateToDiscoverPage = () => {
+  cy.navigateToWorkSpaceHomePage(workspaceName);
+  cy.getElementByTestId('headerAppActionMenu').should('be.visible');
+};
+
+// Sets the dataset for the search. Since INDEXES are not saved to the dataset,
+// we need to click through various buttons to manually add them
+const setDataset = (dataset) => {
+  const datasetType = mapDatasetToType(dataset);
+  cy.getElementByTestId('datasetSelectorButton').should('be.visible').click();
+
+  if (datasetType === 'INDEX_PATTERN') {
+    cy.get(`[title="${dataset}"]`).click();
+  } else if (datasetType === 'INDEXES') {
+    cy.getElementByTestId('datasetSelectorAdvancedButton').click();
+    cy.contains('span', 'Indexes').click();
+    cy.contains('span', datasourceName).click();
+    // this element is sometimes being masked by another element
+    cy.contains('span', dataset).should('be.visible').click({ force: true });
+    cy.getElementByTestId('datasetSelectorNext').click();
+
+    cy.getElementByTestId('advancedSelectorTimeFieldSelect').select('timestamp');
+    cy.getElementByTestId('advancedSelectorConfirmButton').click();
+
+    cy.getElementByTestId('datasetSelectorButton').should(
+      'contain.text',
+      `${datasourceName}::${dataset}`
+    );
+  }
+};
+
+const setDatePickerDatesAndSearchIfRelevant = (language) => {
+  if (language === 'OpenSearch SQL') {
+    return;
+  }
+
+  cy.setTopNavDate(START_TIME, END_TIME);
+};
+
+const setSearchConfigurations = ({
+  addFilter,
+  queryString,
+  setHistogramInterval,
+  selectFields,
+  applySort,
+}) => {
+  if (addFilter) {
+    cy.getElementByTestId('showFilterActions').click();
```

Comment:
should create an utility for addFilter , you can refer to this legacy one https://github.com/opensearch-project/opensearch-dashboards-functional-test/blob/d42d28869c4e11c5f1cad18dbf722f1e8e0c7c1b/cypress/utils/dashboards/data_explorer/commands.js#L74

Reply:
those already existed, so i updated them to work for `isEnhancement`